Student-Service/src/main/java/tw/waterball/judgegirl/studentservice/domain/usecases/GetGroupByIdUseCase.java

```diff
@@ -0,0 +1,34 @@
+package tw.waterball.judgegirl.studentservice.domain.usecases;
+
+import lombok.AllArgsConstructor;
+import tw.waterball.judgegirl.commons.exceptions.NotFoundException;
+import tw.waterball.judgegirl.entities.Group;
+import tw.waterball.judgegirl.studentservice.domain.repositories.GroupRepository;
+
+import javax.inject.Named;
+
+/**
+ * @author - wally55077@gmail.com
+ */
+@Named
+@AllArgsConstructor
+public class GetGroupByIdUseCase {
+
+    private final GroupRepository groupRepository;
+
+    public void execute(Integer groupId, Presenter presenter) throws NotFoundException {
+        validateGroupIdExist(groupId);
+        presenter.setGroup(groupRepository.findGroupById(groupId));
+    }
+
+    private void validateGroupIdExist(Integer id) throws NotFoundException {
+        if (!groupRepository.existsById(id)) {
+            throw new NotFoundException();
+        }
+    }
```

Comment:
You can use `Optional` here !
```java
public void execute(Integer groupId, Presenter presenter) throws NotFoundException {
        presenter.setGroup(groupRepository.findGroupById(groupId)
                     .orElseThrow(NotFoundException::New));
    }
```
But the prerequisite is the `findGroupById` should return `Optional<Group>`.